buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy

```diff
@@ -51,21 +52,36 @@ public class PluginBuildPlugin extends BuildPlugin {
             String name = project.pluginProperties.extension.name
             project.archivesBaseName = name
 
-            // set teh project description so it will be picked up by publishing
+            // set the project description so it will be picked up by publishing
             project.description = project.pluginProperties.extension.description
 
             configurePublishing(project)
 
-            project.integTestCluster.dependsOn(project.bundlePlugin)
+            if (project.plugins.hasPlugin(TestClustersPlugin.class) == false) {
```

Comment:
Performing checks like this is a bit brittle as its subject to plugin application order. This manifests as goofy behavior that's hard to debug. We should try and make plugin interactions immune to application order if at all possible.

Here we could just as well apply the default configuration, then override with `project.pluginManager.withPlugin()` when the test clusters plugin is present.

Reply:
This is inside a `project.afterEvaluate` so it shouldn't depend on the order.

Reply:
Some of them other's aren't  like the one in the rest testing task constructor, so we will need to have testclsuters applied first, but I think that is fine as long as it simplifies setup.  This is not the end state, eventually the plugin build plugin will apply testclusters itself.